### ISSUE TITLE
string type for aria properties

### ISSUE THREAD

**Issue by dgreene1** [dittowords/ditto-react#13]
If we wanted to use aria for alt-text, I think that that the `Ditto` component would not work since it's returning React node type. So given that we're using Typescript, which Ditto component or function returns type string?

The error you get is:
```
(property) ImgHTMLAttributes<HTMLImageElement>.alt?: string | undefined
Type 'Element' is not assignable to type 'string'.(2322)
index.d.ts(2177, 9): The expected type comes from property 'alt' which is declared here on type 'DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>'
```

Here's an example that illustrates the error:
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4BYAKCuADsYkpN0k4BJAEWBhggAUcYAM5wA3lTiS4aXJFpJ6bACYAuAigBGEAK4wAdDG4AbJPjgAfdVt16tSgJ56ZYe2cv4AFriS2IDslQAvlRUMrRC8Fw8EHAAvHAAFGCCQmqc3LwCEMIAlHEAfGISUkQw2lC0cAA8+fgAMkgwIsmNCkrscCh4KHARUHQA5nAwHijwAO4oIqXl8u3YuMMerABSAMoA8gBy+FUA9PlBIZRhEXBKGTHxSSlpUZkpebGF4pRScDMVBA1NcC0MtHabE63V6MH6tCGIzGcEm00asyQ8xweBGq02OwolGC1BOEHC8AAKh5gEIAOrAIxGLYQGBk6AAaziiSeLwkn0qCWK1VAQxQRhgsVEVXuMRk4HxChgyli+E0On0hhgJjMB0CcAOEhyWJxx1ORJJ5J0RiU9KgTOurKKbw5iW5VV5nQFQou0QSonFcilyhUcusiuMpkCOXVmre2qOuKAA

**Comment by azjgard** [comment]
Hi @dgreene1,

Thanks for reporting this.

There's actually no good reason that the `Ditto` component returns `React.ReactNode` instead of `string` -- that just happens to be how it was originally implemented. The component computes the text value and then arbitrarily wraps it in a fragment before being returned.

I'd be happy to remove the fragment that wraps those text values to make this work, but that would technically be a breaking change, so I'm going to earmark including that as part of our next major release.

--------

Underneath the hood, `<Ditto />` and our other components are already using hooks to consume data from `<DittoProvider>` and resolve text. Those hooks haven't yet been exported from the `ditto-react` module because we're not 100% sure what we want the hooks API to look like for library consumers. There probably won't be massive changes from the current form, but we want to get it right before people adopt it.

All that being said, if it would be helpful for you to get ramped up, I can publish a version of `ditto-react` (probably under an `alpha` tag) that has those hooks exported from the module.

This is what passing alt text might look like using a hook:
```jsx
import { useDittoComponent } from 'ditto-react';

const App = () => {
  const text = useDittoComponent({ componentId: 'about.title' });

  return <img alt={text} />
}
```

**Comment by azjgard** [comment]
This will be fixed by #15!
